rpcs3/rpcs3qt/input_dialog.cpp

```diff
@@ -36,7 +40,34 @@ input_dialog::~input_dialog()
 {
 }
 
+void input_dialog::set_clear_button_state(bool enabled)
+{
+	m_input->setClearButtonEnabled(enabled);
+}
+
+void input_dialog::edit_input_font(const QFont& font, bool fix_width, char sample)
+{
+	if (int max = m_input->maxLength(); max > 0 && fix_width && std::isprint(static_cast<uchar>(sample)))
+	{
+		QLabel l(qstr(std::string(static_cast<std::size_t>(max), sample)));
+		l.setFont(font);
+		m_input->setFixedWidth(l.sizeHint().width()); // Set max width based on [font char length * max length] 
```

Comment:
I just remembered, we have this in gui::utils::get_label_width
Maybe you can extend that one or overload it.

Reply:
Added extra QFont ptr arg for it.